src/goto-programs/interpreter.cpp

```diff
@@ -724,15 +730,15 @@ void interpretert::assign(
 void interpretert::execute_assume()
 {
   if(!evaluate_boolean(pc->guard))
-    throw "assumption failed";
+    throw analysis_exceptiont("assumption failed");
```

Comment:
Unrelated to this; I'm not sure this is the right way of handling this case.  I think it should just terminate correctly.